src/main/java/com/mraof/minestuck/block/redstone/VariableSolidSwitchBlock.java

```diff
@@ -0,0 +1,110 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.state.IntegerProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * Can be right clicked without crouching to increase the power/light values the block gives off by one or right clicked with an empty hand to reduce the power/light values by one.
+ * Reaching the highest or lowest value will cause it to cycle back to the opposite extreme
+ */
+public class VariableSolidSwitchBlock extends Block
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	public VariableSolidSwitchBlock(Properties properties)
+	{
+		super(properties);
+		registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	@SuppressWarnings("deprecation")
+	public ActionResultType use(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult hit)
+	{
+		if(!player.isCrouching())
+		{
+			if(state.getValue(POWER) != 15)
```

Comment:
Suggestion: `if(state.getValue(POWER) < 15)`
It shouldn't make any practical difference as the power shouldn't be able to go above 15, but I do think it makes a symbolical difference.